components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -453,17 +470,36 @@ interface ConversationHandler {
   // Send a user-rating for a chat
   // message. |turn_id| is the index of the message in the
   // specified conversation.
-  RateMessage(bool is_liked, uint32 turn_id)
+  RateMessage(bool is_liked, string turn_uuid)
       => (string? rating_id);
   SendFeedback(
       string category,
       string feedback,
       string rating_id, bool send_hostname) => (bool is_success);
 };
 
-interface ConversationEntriesHandler {
+// Browser-side handler for a Conversation's UI responsible for displaying
+// untrusted content (e.g. content generated by the AI engine).
+interface UntrustedConversationHandler {
+  BindUntrustedConversationUI(
+      pending_remote<UntrustedConversationUI> untrusted_ui)
+          => (ConversationEntriesState conversation_entries_state);
+
   // Get all visible history entries, including in-progress responses
   GetConversationHistory() => (array<ConversationTurn> conversation_history);
+
+  ModifyConversation(uint32 turn_index, string new_text);
+};
+
+// Untrusted-UI-side handler for a Conversation, responsible for displaying
+// content generated by the AI engine.
+interface UntrustedConversationUI {
+  // TODO(petemill): Provide single entry that's been updated so that we don't
+  // need to fetch (and clone) all conversation entries each time text is added
+  // to the most recent entry.
+  OnConversationHistoryUpdate();
+  OnEntriesUIStateChanged(ConversationEntriesState state);
+  OnFaviconImageDataChanged();
 };
```

Comment:
This is the API exposed to the untrusted frame from the Conversation

Reply:
not sure if it's a good idea to mix them in the same mojom file? Are there other examples of doing this?